src/classes/CallableApi.cls

```
@@ -0,0 +1,144 @@
+    ********************************************************************************************************/
+    private CallableApiInfo validateAction(String action, Map<String, Object> args) {
+        if (String.isBlank(action)) {
+            return null;
+        }
+
+        String validAction = validActionsMap.get(action.toLowerCase());
+        if (String.isBlank(validAction)) {
+            return null;
+        }
+
+        List<String> actionParams = validAction.split('\\.');
+        if (actionParams.size() == 2) {
+            return new CallableApiInfo(action, actionParams[0], actionParams[1], args);
+        }
+
+        return null;
+    }
+
+    /*******************************************************************************************************
+    * @description Dispatch the action to the appropriate action handlers
+    * @param actionInfo contains the information needed to execute the action
+    * @return Object that can vary depending on the action handler
+    ********************************************************************************************************/
+    private Object dispatchAction(CallableApiInfo actionInfo) {
+        switch on actionInfo.action {
+            when 'TDTM.disableAllTriggers' {
+                TDTM_Config_API.disableAllTriggers();
+            }
+            when else {
+                throw new MalformedMethodInvocationException(actionInfo.action + ' is not implemented');
+            }
+        }
+
+        return generateResponse(actionInfo);
+    }
+
+    /*******************************************************************************************************
+    * @description Generate a generic response to the call for actions that do not have an explicit response
+    ********************************************************************************************************/
+    private Map<String, Object> generateResponse(CallableApiInfo actionInfo) {
```

This heavily limits the return options for our api. The beauty of the callable is that it can return absolutely anything - a List<SObject>, JSON String, Map, etc. This implementation removes all of that capability.

For errors/exceptions, our api or the delegated method should throw the appropriate exception so that it can be handled as needed by the calling package.class.method